pkg/arvo/lib/auto.hoon

```diff
@@ -0,0 +1,265 @@
+::  Autocompletion for hoon.
+::
+|%
++$  ids  (list [=term =type])
+::
+::  Get all the identifiers accessible if this type is your subject.
+::
+++  get-identifiers
+  |=  ty=type
+  %-  flop
+  |-  ^-  ids
+  ?-    ty
+      %noun      ~
+      %void      ~
+      [%atom *]  ~
+      [%cell *]
+    %+  weld
+      $(ty p.ty)
+    $(ty q.ty)
+  ::
+      [%core *]
+    %-  weld
+    :_  $(ty p.ty)
```

Comment:
Do you think you should limit the results by checking the variance here?

Reply:
Good catch, I think you're right.

Reply:
Beautiful:

```
-----
de  {a/@ b/@} -> u(@)
en  {a/@ b/@} -> @
nu  <3?kyj 6.zxs 33.rft 27.ipf 7.ecf 36.uek 92.bjk 247.ows 51.mvt 126.xjf 41.mac 1.ane>
ex  <4?zto 6.zxs 33.rft 27.ipf 7.ecf 36.uek 92.bjk 247.ows 51.mvt 126.xjf 41.mac 1.ane>
dy  {a/@ b/@} -> @
as  <4?erk 6.zxs 33.rft 27.ipf 7.ecf 36.uek 92.bjk 247.ows 51.mvt 126.xjf 41.mac 1.ane>
~zod:dojo> :*acru:ames
```

Reply:
`[tab pos=0]`

Reply:
now just give those samples meaningful names and we can get rid of the docs altogether!

Reply:
now just give those samples meaningful names and we can get rid of the docs altogether!